common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -137,6 +143,37 @@ public TTransportFactory getServerTransportFactory() throws SaslException {
     }
   }
 
-  // TODO: get client side TTransport based on auth type
-  // public TTransport getClientTransport(InetSocketAddress serverAddress) throws SaslException {}
+  /**
+   * Create a transport per the connection options. Supported transport options are: NOSASL,
+   * SIMPLE, CUSTOM, KERBEROS
```

Comment:
My understanding is, with `NOSASL` as input, an unmodified `TTransport` is returned; with `SIMPLE` as input, a `PlainCLientTransport` is returned, correct? It is better to javadoc those differences, just to be more explicit.